lib/chef_zero/endpoints/cookbook_artifacts_cookbook_identifier.rb

```diff
@@ -0,0 +1,69 @@
+require 'chef_zero/chef_data/data_normalizer'
+
+module ChefZero
+  module Endpoints
+    class CookbookArtifactsCookbookIdentifierEndpoint < RestBase
+      # these endpoints are almost, but not quite, not entirely unlike the corresponding /cookbooks endpoints.
+      # it could all be refactored for maximum reuse, but they're short REST methods with well-defined
+      # behavioral specs (pedant), so there's not a huge benefit.
+
+      # GET /organizations/ORG/cookbook_artifacts/NAME/IDENTIFIER
+      def get(request)
+        cookbook_data = normalize(request, parse_json(get_data(request)))
+        return json_response(200, cookbook_data)
+      end
+
+      # PUT /organizations/ORG/cookbook_artifacts/COOKBOOK/IDENTIFIER
+      def put(request)
+        if exists_data?(request)
+          return error(409, "Cookbooks cannot be modified, and a cookbook with this identifier already exists.")
+        end
+
+        set_data(request, nil, request.body, :create_dir)
+
+        return already_json_response(201, request.body)
+      end
+
+      # DELETE /organizations/ORG/cookbook_artifacts/COOKBOOK/IDENTIFIER
+      def delete(request)
+        begin
+          identified_cookbook_data = normalize(request, parse_json(get_data(request)))
+          delete_data(request)
+
+          # go through the recipes and delete stuff in the file store.
+          identified_cookbook_data["recipes"].each do |r|
```

Comment:
I would expect there to be more than recipes here. Shouldn't it go through all segments of the cookbook files?